backend/src/main/kotlin/com/group7/artshare/controller/ProfileController.kt

```diff
@@ -26,5 +31,53 @@ class ProfileController(
         return profileService.getUserByUsernameOrToken(null, authorizationHeader)
     }
 
+    @GetMapping("settings")
+    fun getSettingsForUser(
+        @RequestHeader(
+            value = "Authorization",
+            required = true
+        ) authorizationHeader: String?
+    ) : SettingDTO {
+        try {
+            authorizationHeader?.let {
+                val user =
+                    jwtService.getUserFromAuthorizationHeader(authorizationHeader) ?: throw Exception("Invalid token")
+                return profileService.getSettings(user)
+            } ?: throw ResponseStatusException(HttpStatus.BAD_REQUEST, "Unregistered user cannot view user settings")
+        } catch (e: Exception) {
+            if (e.message == "Invalid token") {
+                throw ResponseStatusException(HttpStatus.UNAUTHORIZED, e.message)
+            } else {
+                throw ResponseStatusException(HttpStatus.BAD_REQUEST, e.message)
+            }
+        }
+    }
+
+    @PostMapping(
+        value = ["settings"],
+        consumes = ["application/json;charset=UTF-8"],
+        produces = ["application/json;charset=UTF-8"]
+    )
+    fun setSettingsForUser(
+        @RequestBody setting : SettingDTO,
+        @RequestHeader(
+            value = "Authorization",
+            required = true
+        ) authorizationHeader: String?
```

Comment:
please remove optional part of String? since it is required. You can look at this [PR](https://github.com/bounswe/bounswe2022group7/pull/434)'s corresponding part as well